rust/flexbuffers/src/builder/push.rs

```diff
@@ -0,0 +1,167 @@
+        impl Sealed for $T {}
+        impl Pushable for $T {
+            fn push_to_builder(self, builder: &mut Builder) {
+                builder.$method(self as $asT);
+            }
+        }
+    };
+}
+forward_to_builder!(&str, push_str);
+forward_to_builder!(bool, push_bool);
+forward_to_builder!(u8, push_uint);
+forward_to_builder!(u16, push_uint);
+forward_to_builder!(u32, push_uint);
+forward_to_builder!(u64, push_uint);
+forward_to_builder!(i8, push_int);
+forward_to_builder!(i16, push_int);
+forward_to_builder!(i32, push_int);
+forward_to_builder!(i64, push_int);
+forward_to_builder!(f32, push_float);
+forward_to_builder!(f64, push_float);
+forward_to_builder!(&[u8], push_uints);
+forward_to_builder!(&[u16], push_uints);
+forward_to_builder!(&[u32], push_uints);
+forward_to_builder!(&[u64], push_uints);
+forward_to_builder!(&[i8], push_ints);
+forward_to_builder!(&[i16], push_ints);
+forward_to_builder!(&[i32], push_ints);
+forward_to_builder!(&[i64], push_ints);
+forward_to_builder!(&[f32], push_floats);
+forward_to_builder!(&[f64], push_floats);
+forward_to_builder!(&[bool], push_bools);
+forward_to_builder!(&Vec<u8>, push_uints);
+forward_to_builder!(&Vec<u16>, push_uints);
+forward_to_builder!(&Vec<u32>, push_uints);
+forward_to_builder!(&Vec<u64>, push_uints);
+forward_to_builder!(&Vec<i8>, push_ints);
+forward_to_builder!(&Vec<i16>, push_ints);
+forward_to_builder!(&Vec<i32>, push_ints);
+forward_to_builder!(&Vec<i64>, push_ints);
+forward_to_builder!(&Vec<f32>, push_floats);
+forward_to_builder!(&Vec<f64>, push_floats);
+forward_to_builder!(&Vec<bool>, push_bools);
+
+macro_rules! impl_indirects {
+    ($Indirect: ident, $method: ident) => {
+        impl Sealed for $Indirect {}
+        impl Pushable for $Indirect {
+            fn push_to_builder(self, builder: &mut Builder) {
+                builder.$method(self.0);
+            }
+        }
+    };
+}
+impl_indirects!(IndirectInt, push_indirect_int);
+impl_indirects!(IndirectUInt, push_indirect_uint);
+impl_indirects!(IndirectFloat, push_indirect_float);
+
+macro_rules! impl_arrays {
+    ($num: expr) => {
+        forward_to_builder!(&[u8; $num], push_uints, &[u8]);
+        forward_to_builder!(&[u16; $num], push_uints, &[u16]);
+        forward_to_builder!(&[u32; $num], push_uints, &[u32]);
+        forward_to_builder!(&[u64; $num], push_uints, &[u64]);
+        forward_to_builder!(&[i8; $num], push_ints, &[i8]);
+        forward_to_builder!(&[i16; $num], push_ints, &[i16]);
+        forward_to_builder!(&[i32; $num], push_ints, &[i32]);
+        forward_to_builder!(&[i64; $num], push_ints, &[i64]);
+        forward_to_builder!(&[f32; $num], push_floats, &[f32]);
+        forward_to_builder!(&[f64; $num], push_floats, &[f64]);
+        forward_to_builder!(&[bool; $num], push_bools, &[bool]);
+    };
+}
+impl_arrays!(0);
+impl_arrays!(1);
+impl_arrays!(2);
+impl_arrays!(3);
+impl_arrays!(4);
+impl_arrays!(5);
+impl_arrays!(6);
```

Comment:
this doesn't seem great.. but I am probably missing the Rust details..

Reply:
Yep this sucks. We can't be generic over rust fixed length arrays until [const generics](https://github.com/rust-lang/rust/issues/44580) are a thing.

Alternatively,  we can scrap the overloaded `push` and have `push_uint` `push_blob` etc. Upsides: Its more explicit and probably has faster compile times.
Downsides: Its more verbose.

I could go either way on this one tbh

Reply:
@rw do you have an opinion on this?

Reply:
@CasperN I like that you used some of the idioms from the actual flatbuffers crate. Unfortunately, I don't see an immediately-better thing to do. If you include these in the catchall structs I asked for for quickcheck, then I'm fine with letting these implementations sit as-is until const generics are stabilized. We can refactor this later.

Reply:
Check out qc_serious.rs for a catchall quickcheck

Reply:
@rw How does that look for you?